docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/index.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: docs
+title: Custom Browser Provider Plugin
+permalink: /documentation/extending-testcafe/custom-browser-provider-plugin/
+---
+# Custom Browser Provider Plugin
+
+TestCafe has built-in support for testing in a number of popular browsers (see [Using Browsers](../../using-testcafe/common-concepts/using-browsers.md)).
+If you need to use a different browser (for example, a browser from a cloud service), you can create a **custom browser provider plugin**.
+The provider should expose methods for performing common actions on the browser windows during testing: opening and closing it, taking screenshot, etc.
+
+To create a browser provider plugin, go through the following steps.
+
+* [Generating Browser Provider Project](#generating-browser-provider-project)
+* [Implementing the Browser Provider](#implementing-the-browser-provider)
+* [Building the Provider](#building-the-provider)
+* [Using the Provider Development Version](#using-the-provider-development-version)
+* [Publishing the Provider to NPM](#publishing-the-provider-to-npm)
+* [Reference](#reference)
```

Comment:
I suppose this is a mistake:
_... go through the following steps._

_-generating_
_-implementing_
_-building_
_-using_
_-publishing_
**-reference**

_Reference_ should not be here?